fundingmanager.go

```diff
@@ -1829,82 +1800,50 @@ func (f *fundingManager) handleFundingSigned(fmsg *fundingSignedMsg) {
+// funding broadcast height. In case of confirmation, the short channel ID of
+// the channel will be returned.
+func (f *fundingManager) waitForFundingWithTimeout(
+	ch *channeldb.OpenChannel) (*lnwire.ShortChannelID, error) {
 
-	epochClient, err := f.cfg.Notifier.RegisterBlockEpochNtfn(nil)
-	if err != nil {
-		fndgLog.Errorf("unable to register for epoch notification: %v",
-			err)
-		close(confChan)
-		return
-	}
-
-	defer epochClient.Cancel()
-
-	waitingConfChan := make(chan *lnwire.ShortChannelID)
+	confChan := make(chan *lnwire.ShortChannelID)
+	timeoutChan := make(chan error, 1)
 	cancelChan := make(chan struct{})
 
-	// Add this goroutine to wait group so we can be sure that it is
-	// properly stopped before the funding manager can be shut down.
 	f.wg.Add(1)
-	go f.waitForFundingConfirmation(completeChan, cancelChan,
-		waitingConfChan)
-
-	// On block maxHeight we will cancel the funding confirmation wait.
-	maxHeight := completeChan.FundingBroadcastHeight + maxWaitNumBlocksFundingConf
-	for {
-		select {
-		case epoch, ok := <-epochClient.Epochs:
-			if !ok {
-				fndgLog.Warnf("Epoch client shutting down")
-				return
-			}
-
-			// If we are not the channel initiator it's safe
-			// to timeout the channel
-			if uint32(epoch.Height) >= maxHeight && !completeChan.IsInitiator {
-				fndgLog.Warnf("waited for %v blocks without "+
-					"seeing funding transaction confirmed,"+
-					" cancelling.", maxWaitNumBlocksFundingConf)
-
-				// Cancel the waitForFundingConfirmation
-				// goroutine.
-				close(cancelChan)
+	go f.waitForFundingConfirmation(ch, cancelChan, confChan)
 
-				// Notify the caller of the timeout.
-				close(timeoutChan)
-				return
-			}
+	// If we are not the initiator, we have no money at stake and will
+	// timeout waiting for the funding transaction to confirm after a
+	// while.
+	if !ch.IsInitiator {
+		f.wg.Add(1)
+		go f.waitForTimeout(ch, cancelChan, timeoutChan)
+	}
+	defer close(cancelChan)
 
-			// TODO: If we are the channel initiator implement
-			// a method for recovering the funds from the funding
-			// transaction
+	var shortChanID *lnwire.ShortChannelID
+	var ok bool
+	select {
+	case err := <-timeoutChan:
+		if err != nil {
+			return nil, err
+		}
+		return nil, ErrConfirmationTimeout
 
-		case <-f.quit:
-			// The fundingManager is shutting down, will resume
-			// waiting for the funding transaction on startup.
-			return
-		case shortChanID, ok := <-waitingConfChan:
-			if !ok {
-				// Failed waiting for confirmation, close
-				// confChan to indicate failure.
-				close(confChan)
-				return
-			}
+	case <-f.quit:
+		// The fundingManager is shutting down, and will resume wait on
+		// startup.
+		return nil, ErrFundingManagerShuttingDown
 
-			select {
-			case confChan <- shortChanID:
-			case <-f.quit:
-				return
-			}
+	case shortChanID, ok = <-confChan:
+		if !ok {
+			return nil, fmt.Errorf("Waiting for funding" +
```

Comment:
Nit: error not logged should start lowercase.

Reply:
Done: https://github.com/lightningnetwork/lnd/pull/1910/commits/e148c0a1af38e6b4565f666261cec638c27ddebe